src/fix_bd_sphere.cpp

```diff
@@ -0,0 +1,303 @@
+	noise_flag = 1;
+      } else if (strcmp(arg[iarg + 1],"gaussian") == 0) {
+	noise_flag = 1;
+	gaussian_noise_flag = 1;
+      } else if (strcmp(arg[iarg + 1],"none") == 0) {
+	noise_flag = 0;
+      } else {
+	error->all(FLERR,"Illegal fix/bd/sphere command.");
+      }
+    } else {
+      error->all(FLERR,"Illegal fix/bd/sphere command.");
+    }
+    iarg = iarg + 2;
+  }
+
+  
+  // initialize Marsaglia RNG with processor-unique seed
+  random = new RanMars(lmp,seed + comm->me);
+
+
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixBdSphere::setmask()
+{
+  int mask = 0;
+  mask |= INITIAL_INTEGRATE;
+  mask |= POST_FORCE;
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+FixBdSphere::~FixBdSphere()
+{
+
+  delete random;
+
+}
+
+
+
+/* ---------------------------------------------------------------------- */
+
+void FixBdSphere::init()
+{
+
+
+  g1 =  force->ftm2v/gamma_t;
+  g3 = force->ftm2v/gamma_r;
+  if (noise_flag == 0) {
+    g2 = 0;
+    g4 = 0;
+    rng_func = &RanMars::zero_rng;
+  } else if (gaussian_noise_flag == 1) {
+    g2 = gamma_t*sqrt(2 * diff_t)/force->ftm2v;
+    g4 = gamma_r*sqrt(2 * diff_r)/force->ftm2v;
+    rng_func = &RanMars::gaussian;
+  } else {
+    g2 = gamma_t*sqrt( 24 * diff_t)/force->ftm2v;
+    g4 = gamma_r*sqrt( 24 * diff_r )/force->ftm2v;
+    rng_func = &RanMars::uniform_middle;
+  }
+
+  dt = update->dt;
+  sqrtdt = sqrt(dt);
+}
+
+void FixBdSphere::setup(int vflag)
+{
+  post_force(vflag);
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixBdSphere::initial_integrate(int /* vflag */)
+{
+  double **x = atom->x;
+  double **v = atom->v;
+  double **mu = atom->mu;
+  double **f = atom->f;
+  double **omega = atom->omega;
+  double **torque = atom->torque;
+  int *mask = atom->mask;
+  int nlocal = atom->nlocal;
+  double dx,dy,dz;
+  double dtheta;
+  double mux,muy,muz,mu_tmp,wx,wy,wz;
+  double prefac_1, prefac_2;
+  
+  if (igroup == atom->firstgroup) nlocal = atom->nfirst;
+
+  int d3rot;  // whether to compute angular momentum in xy plane
+
+  dt = update->dt;
+  sqrtdt = sqrt(dt);
+  
+  if (domain->dimension==2) {
+    d3rot = 0;
+  } else {
+    d3rot = 1;
+  }
+
+  
+  for (int i = 0; i < nlocal; i++) {
+    if (mask[i] & groupbit) {
+      
+      dx = dt * g1 * f[i][0];
+      x[i][0] +=  dx;
+      v[i][0]  =  dx/dt;
+      
+      dy = dt * g1 * f[i][1];
+      x[i][1] +=  dy;
+      v[i][1]  =  dy/dt;
+      
+      dz = dt * g1 * f[i][2];
+      x[i][2] +=  dz;
+      v[i][2]  =  dz/dt;
+      
+      
+      omega[i][0] = d3rot * g3* torque[i][0];
+      omega[i][1] = d3rot * g3* torque[i][1];
+      omega[i][2] = g3* torque[i][2];
+      
+      dtheta = sqrt((omega[i][0]*dt)*(omega[i][0]*dt)
+		    +(omega[i][1]*dt)*(omega[i][1]*dt)
+		    +(omega[i][2]*dt)*(omega[i][2]*dt));
+      
+      if (abs(dtheta) < 1e-14) {
```

Comment:
please use a global constant or define (usually called SMALL) .
also `abs()` applies to integers, and thus you should be using `fabs()` here.
please also use correct indentation below and note that there should be no tabs, only blanks.

Reply:
Okay thank you!